test/services/VatPartialBuilderSpec.scala

```diff
@@ -239,6 +258,61 @@ class VatPartialBuilderSpec extends ViewSpecBase with OneAppPerSuite with Mockit
         )
       }
 
+      "the user is in debit and has a Direct Debit set up" in new PaymentsSetup {
+        val enrolmentStore: testEnrolmentsStoreService = new testEnrolmentsStoreService(false)
+        override lazy val accountBalance = AccountBalance(Some(BigDecimal(12.34)))
+        override val vatData = VatData(accountSummaryData.copy(openPeriods = openPeriods), Some(calendarWithDirectDebit))
+
+        val view: String = new VatPartialBuilderImpl(enrolmentStore, emacUrlBuilder, config).buildPaymentsPartial(vatData)(fakeRequestWithEnrolments, messages).body
+        val doc: Document = Jsoup.parse(view)
+
+        doc.text().contains("You owe £12.34") mustBe true
+        doc.text().contains("You have a VAT Direct Debit. If you complete your return on time, we will take payment for the period ending 30 June 2016 on 15 August 2016. You can change or cancel your Direct Debit.") mustBe true
+
+        assertLinkById(
+          doc,
+          linkId = "vat-change-cancel-direct-debit",
+          expectedText = "change or cancel your Direct Debit",
+          expectedUrl = "http://localhost:9733/business-account/help/vat/how-to-pay",
+          expectedGAEvent = "link - click:VAT cards:Change or cancel your direct debit",
+          expectedIsExternal = false,
+          expectedOpensInNewTab = false
+        )
+
+        assertLinkById(
+          doc,
+          linkId = "vat-make-payment-link",
+          expectedText = "Make a VAT payment",
+          expectedUrl = "http://localhost:9732/business-account/vat/make-a-payment",
+          expectedGAEvent = "link - click:VAT cards:Make a VAT payment",
+          expectedIsExternal = false,
+          expectedOpensInNewTab = false
+        )
+      }
+
+
+      "the user is in debit and files annually (should not see DD)" in new PaymentsSetup {
+        val enrolmentStore: testEnrolmentsStoreService = new testEnrolmentsStoreService(false)
+        override lazy val accountBalance = AccountBalance(Some(BigDecimal(12.34)))
+        override val vatData = VatData(accountSummaryData.copy(openPeriods = openPeriods), Some(calendarWithAnnualFiling))
+
+        val view: String = new VatPartialBuilderImpl(enrolmentStore, emacUrlBuilder, config).buildPaymentsPartial(vatData)(fakeRequestWithEnrolments, messages).body
+        val doc: Document = Jsoup.parse(view)
+
+        doc.text().contains("You owe £12.34") mustBe true
+        doc.text().contains("Direct Debit") mustBe false
+
+        assertLinkById(
+          doc,
+          linkId = "vat-make-payment-link",
+          expectedText = "Make a VAT payment",
+          expectedUrl = "http://localhost:9732/business-account/vat/make-a-payment",
+          expectedGAEvent = "link - click:VAT cards:Make a VAT payment",
+          expectedIsExternal = false,
+          expectedOpensInNewTab = false
+        )
+      }
+
```

Comment:
I'd like to see unit tests for the inactive DD and ineligible DD (for reasons other than annual filing) cases

Reply:
There's already a test for InactiveDirectDebit. I added one for Ineligible.